engine/item_effects.asm

```diff
@@ -2313,6 +2316,37 @@ ItemWasntUsedMessage:
 	jp PrintText
 ; f7e8
 
+Ball_ReplacePartyMonCaughtBall:
+	ld b, PARTYMENUACTION_00
+	call UseItem_SelectMon
+	jp c, ItemNotUsed_ExitMenu
+
+	ld a, [wCurItem]
+	ld b, a
+	ld a, [wCurPartyMon]
+	ld hl, wPartyMon1CaughtBall
+	call GetPartyLocation
+	ld a, [hl]
+	and CAUGHTBALL_MASK
+	cp b
+	
+	jp z, AlreadyInThatBallMessage
+	ld a, [hl]
+	and -CAUGHTBALL_MASK-1 ; One's complement
```

Comment:
The typical way to write this is `and $ff ^ CAUGHTBALL_MASK` (or sometimes `and $ff - CAUGHTBALL_MASK` from before rgbasm handled byte xor correctly).